src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -119,7 +123,7 @@ def _create_node_task(
             task = self._sync_function_to_async_task(executor, context, node, f, kwargs)
         # Set the name of the task to the node name for debugging purpose
         # It does not need to be unique by design.
-        return asyncio.create_task(task, name=node.name)
+        return asyncio.create_task(self.run_task_with_semaphore(task), name=node.name)
```

Comment:
It seems that the task is created before we put it to the semaphore, we may add more detailed comments why this works:
1 For async task, does it mean that the the task will not be executed before calling `create_task`?
2 For sync task, does it mean that even though the task is put to the threading pool, thread.start will only be triggered after `create_task`? 

It would be better to confirm these details and add more comments here.

Reply:
After test, we can confirm that the task will not run until calling create_task. Added comments explaining the calling sequence in _create_node_task.
![1705384853363](https://github.com/microsoft/promptflow/assets/39176492/7dc6a383-5bdb-4cfb-a421-9c9689522db1)